tests/acceptance/test_v2_measurement_service.py

```diff
@@ -0,0 +1,187 @@
+    data_size: int, stub_v2: v2_measurement_service_pb2_grpc.MeasurementServiceStub
+):
+    """End to End Test validating measurement returns expected amount of cumulatively data."""
+    name = "test"
+    num_responses = 10
+    cumulative_data = True
+    response_interval_in_ms = 1
+    error_on_index = -1
+
+    request = v2_measurement_service_pb2.MeasureRequest(
+        configuration_parameters=_get_configuration_parameters(
+            name, num_responses, data_size, cumulative_data, response_interval_in_ms, error_on_index
+        )
+    )
+
+    response_iterator = stub_v2.Measure(request)
+
+    expected_data: List[int] = []
+    index = 0
+    for response in response_iterator:
+        expected_data.extend(index for i in range(data_size))
+        expected = _get_serialized_measurement_outputs(name, index, expected_data)
+        assert expected == response.outputs.value
+        index += 1
+
+    assert index == num_responses
+
+
+@pytest.mark.parametrize("data_size", [1, 10, 100])
+def test___streaming_measurement_service___specify_data_size___receives_expected_amount_of_data(
+    data_size: int, stub_v2: v2_measurement_service_pb2_grpc.MeasurementServiceStub
+):
+    """End to End Test validating measurement returns the expected amount of non-cumulative data."""
+    name = "test"
+    num_responses = 10
+    cumulative_data = False
+    response_interval_in_ms = 1
+    error_on_index = -1
+
+    request = v2_measurement_service_pb2.MeasureRequest(
+        configuration_parameters=_get_configuration_parameters(
+            name, num_responses, data_size, cumulative_data, response_interval_in_ms, error_on_index
+        )
+    )
+
+    response_iterator = stub_v2.Measure(request)
+
+    index = 0
+    for response in response_iterator:
+        expected_data = [index for i in range(data_size)]
+        expected = _get_serialized_measurement_outputs("test", index, expected_data)
+        assert expected == response.outputs.value
+        index += 1
+
+    assert index == num_responses
+
+
+@pytest.mark.parametrize("error_on_index", [1, 5, 9])
+def test___streaming_measurement_service___specify_error_index___errors_at_expected_response(
+    error_on_index: int, stub_v2: v2_measurement_service_pb2_grpc.MeasurementServiceStub
+):
+    """End to End Test validating streaming measurement sends responses prior to expected error."""
+    name = "test"
+    num_responses = 10
+    data_size = 1
+    cumulative_data = True
+    response_interval_in_ms = 1
+    request = v2_measurement_service_pb2.MeasureRequest(
+        configuration_parameters=_get_configuration_parameters(
+            name, num_responses, data_size, cumulative_data, response_interval_in_ms, error_on_index
+        )
+    )
+
+    response_iterator = stub_v2.Measure(request)
+
+    index = 0
+    with pytest.raises(grpc.RpcError):
+        for response in response_iterator:
+            index += 1
+
+    assert index == error_on_index
```

Comment:
You could also check the index in the error message.
```suggestion
    with pytest.raises(grpc.RpcError) as exc_info:
        for response in response_iterator:
            index += 1

    assert index == error_on_index
    assert f"Errored at index: {error_on_index}\n" in exc_info.value.args[0]
```